blocks/header/header.js

```diff
@@ -85,6 +89,107 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+async function fetchSearchData() {
+  if (!window.searchData) {
+    const resp = await fetch(`/${window.location.pathname.split('/')[1]}/query-index.json`);
+    const json = await resp.json();
+    window.searchData = json;
+  }
+  return window.searchData;
+}
+
+async function search(value) {
+  const searchData = await fetchSearchData();
+  const hits = [];
+  for (let i = 0; i < searchData.data.length; i += 1) {
+    const e = searchData.data[i];
+    const text = [e.title, e.description].join(' ').toLowerCase();
+    if (text.includes(value.toLowerCase())) {
+      hits.push(e);
+    }
+  }
+  return hits;
+}
+
+function decorateSearch(block) {
+  const searchSection = block.querySelector('a > .icon-search');
```

Comment:
i would be slightly more specific, e.g. with `header a > .icon-search` - just in case that a search icon figures somewhere else on the site.